src/core/server/saved_objects/migrationsv2/integration_tests/multiple_es_nodes.test.ts

```diff
@@ -0,0 +1,211 @@
+import * as kbnTestServer from '../../../../test_helpers/kbn_server';
+import type { ElasticsearchClient } from '../../../elasticsearch';
+import { Root } from '../../../root';
+
+const LOG_FILE_PREFIX = 'migration_test_multiple_es_nodes';
+
+const asyncUnlink = Util.promisify(Fs.unlink);
+
+async function removeLogFile() {
+  glob(Path.join(__dirname, `${LOG_FILE_PREFIX}_*.log`), (err, files) => {
+    files.forEach(async (file) => {
+      // ignore errors if it doesn't exist
+      await asyncUnlink(file).catch(() => void 0);
+    });
+  });
+}
+
+function extractSortNumberFromId(id: string): number {
+  const parsedId = parseInt(id.split(':')[1], 10); // "foo:123" -> 123
+  if (isNaN(parsedId)) {
+    throw new Error(`Failed to parse Saved Object ID [${id}]. Result is NaN`);
+  }
+  return parsedId;
+}
+
+async function fetchDocs(esClient: ElasticsearchClient, index: string, type: string) {
+  const { body } = await esClient.search<any>({
+    index,
+    size: 10000,
+    body: {
+      query: {
+        bool: {
+          should: [
+            {
+              term: { type },
+            },
+          ],
+        },
+      },
+    },
+  });
+
+  return body.hits.hits
+    .map((h) => ({
+      ...h._source,
+      id: h._id,
+    }))
+    .sort((a, b) => extractSortNumberFromId(a.id) - extractSortNumberFromId(b.id));
+}
+
+interface RootConfig {
+  logFileName: string;
+  hosts: string[];
+}
+
+function createRoot({ logFileName, hosts }: RootConfig) {
+  return kbnTestServer.createRoot({
+    elasticsearch: {
+      hosts,
+      username: kibanaServerTestUser.username,
+      password: kibanaServerTestUser.password,
+    },
+    migrations: {
+      skip: false,
+      enableV2: true,
+      batchSize: 100, // fixture contains 5000 docs
+    },
+    logging: {
+      appenders: {
+        file: {
+          type: 'file',
+          fileName: logFileName,
+          layout: {
+            type: 'json',
+          },
+        },
+      },
+      loggers: [
+        {
+          name: 'root',
+          appenders: ['file'],
+        },
+      ],
+    },
+  });
+}
+
+describe('migration v2', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let root: Root;
+  const migratedIndex = `.kibana_${pkg.version}_001`;
+
+  beforeAll(async () => {
+    await removeLogFile();
+  });
+
+  afterAll(async () => {
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  });
+
+  afterEach(async () => {
+    if (root) {
+      await root.shutdown();
+    }
+
+    if (esServer) {
+      await esServer.stop();
+    }
+  });
+
+  it('migrates saved objects normally with multiple ES nodes', async () => {
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'basic',
+          port: 9220,
+          clusterName: 'es-test-cluster',
+          esArgs: ['xpack.security.enabled=false'],
```

Comment:
I had disabled security during experimentation and forgot to remove it, only to discover that the new multi-node settings in `createTestEsCluster` seem to not work when security is enabled (single-node clusters still seem to work fine).

The test here should pass as-is, but I am working to sort this out so the updated test utils can be used as generically as possible.

Reply:
Fixed, we were automatically doing native realm setup when each node was started, causing failures if the cluster wasn't formed yet.